File: Task 24/ClearHire-Job Portal/Backend/app/crud/companies.py
from sqlalchemy.orm import Session

from app.models.company import Company


def create_company(
    db: Session,
    user_id: int,
    name: str,
    description: str | None,
    website: str | None,
    location: str | None,
    industry: str | None,
):
    company = Company(
        user_id=user_id,
        name=name,
        description=description,
        website=website,
        location=location,
        industry=industry,
    )

    db.add(company)
    db.commit()
    db.refresh(company)

    return company


def get_company(
    db: Session,
    company_id: int,
):
    return (
        db.query(Company)
        .filter(Company.id == company_id)
        .first()
    )


def get_company_by_user(
    db: Session,
    user_id: int,
):
    return (
        db.query(Company)
        .filter(Company.user_id == user_id)
        .first()
    )


def get_companies(
    db: Session,
):
    return db.query(Company).all()


def update_company(
    db: Session,
    company: Company,
    data: dict,
):
    for field, value in data.items():
        setattr(company, field, value)

    db.commit()
    db.refresh(company)

    return company


def delete_company(
    db: Session,
    company: Company,
):
    db.delete(company)
    db.commit()

File: Task 25/ClearHire-Job Portal/Frontend/frontend/src/services/companies.js
import api from "./api";

export const createCompany = async (companyData) => {
  const response = await api.post("/companies", companyData);
  return response.data;
};

export const getCompanies = async () => {
  const response = await api.get("/companies");
  return response.data;
};

export const getCompany = async (companyId) => {
  const response = await api.get(`/companies/${companyId}`);
  return response.data;
};

export const updateCompany = async (companyId, companyData) => {
  const response = await api.put(`/companies/${companyId}`, companyData);
  return response.data;
};

export const deleteCompany = async (companyId) => {
  await api.delete(`/companies/${companyId}`);
};